src/twitter.ts

```diff
@@ -111,44 +100,39 @@ const withCache = async <T, U>(
     }
   }
 
+  // Refresh cache if miss
   const value = await producer();
   await redis.set(key, JSON.stringify(value));
   return callback(value);
 };
 
-type TFetchTweetsSuccess = {
-  ok: true;
-  tweets: any;
-};
-
-type TFetchTweetsUserDoesNotExist = {
-  ok: false;
-  error: 'User does not exist';
-};
-
+// Types for fetch tweets response
+type TFetchTweetsSuccess = { ok: true; tweets: any; };
+type TFetchTweetsUserDoesNotExist = { ok: false; error: 'User does not exist'; };
 type TFetchTweetsResult = TFetchTweetsSuccess | TFetchTweetsUserDoesNotExist;
 
-export const fetchTweets = async (redis: RedisClient, username: string): Promise<TFetchTweetsResult> =>
-  withCache({
+// Main function to fetch tweets
```

Comment:
Personally, I wouldn't call this the main function, since there might be other functions added in the future that exist alongside it. I think the function name speaks for itself here and doesn't need explaining.